### PR TITLE
Bug 1384746 added section on tuning sysctl values

### PR DESCRIPTION
**Pull request by bfallonf**
As per: https://bugzilla.redhat.com/show_bug.cgi?id=1384746

@jeremyeder @jmencak @ahardin-rh as per email.

I changed around some of the wording. Any thoughts?

### CODE REVIEW

**Comment by jmencak** [comment]
Rewording seems good to me.

File: install_config/router/default_haproxy_router.adoc [hunk 1]
```diff
@@ -1122,6 +1122,44 @@ $ oc set env dc/router -c router DROP_SYN_DURING_RESTART=true
...
+threshold, increase the values. The following sysctl values are recommended for large-scale clusters:
+
+----
+net.ipv4.neigh.default.gc_thresh1 = 8192
+net.ipv4.neigh.default.gc_thresh2 = 32768
+net.ipv4.neigh.default.gc_thresh3 = 65536
+----
+
+To make these settings permanent across reboots, create a 
+https://access.redhat.com/solutions/1305833[custom tuned profile].
```
Comment (ahardin-rh) [inline install_config/router/default_haproxy_router.adoc]: link:https://access.redhat.com/solutions/1305833[custom tuned profile]

**Comment by ahardin-rh** [comment]
Just one comment. Otherwise, LGTM. Thanks!

File: install_config/router/default_haproxy_router.adoc [hunk 2]
```diff
@@ -1122,6 +1122,44 @@ $ oc set env dc/router -c router DROP_SYN_DURING_RESTART=true
...
 
+[[deploy-router--arp-cach-tuning-for-large-scale-clusters]]
+== ARP Cache Tuning for Large-scale Clusters
+
+In {product-title}} clusters with large numbers of routes (greater than the
+value of `net.ipv4.neigh.default.gc_thresh3`, which is `1024` by default), you
+must increase the default values of sysctl variables to allow more entries in
+the ARP cache.
+
+In clusters, the kernel messages would be similar to the following:
```
Comment (jeremyeder) [inline install_config/router/default_haproxy_router.adoc]: s/In clusters, t/T

File: install_config/router/default_haproxy_router.adoc [hunk 3]
```diff
@@ -1122,6 +1122,44 @@ $ oc set env dc/router -c router DROP_SYN_DURING_RESTART=true
...
+
+----
+[ 1738.811139] net_ratelimit: 1045 callbacks suppressed
+[ 1743.823136] net_ratelimit: 293 callbacks suppressed
+----
+
+When using the `oc` client, the commands fail and the user receives the following:
```
Comment (jeremyeder) [inline install_config/router/default_haproxy_router.adoc]: When this issue occurs, you may begin to see the `oc` client commands being to fail with the following error:

File: install_config/router/default_haproxy_router.adoc [hunk 4]
```diff
@@ -1122,6 +1122,44 @@ $ oc set env dc/router -c router DROP_SYN_DURING_RESTART=true
...
+----
+
+To verify the actual amount of ARP entries for IPv4, run the following:
+
+----
+# ip -4 neigh show nud all | wc -l
+----
+
+If the number begins to approach the `net.ipv4.neigh.default.gc_thresh3`
+threshold, increase the values. The following sysctl values are recommended for large-scale clusters:
```
Comment (jeremyeder) [inline install_config/router/default_haproxy_router.adoc]: s/recommended/suggested for clusters with large numbers of routes.

Note: we should state which machine this needs to be run on...nodes running router pods only, right?  So, most likely the infra nodes?

**Review by jeremyeder — CHANGES_REQUESTED** [review]

File: install_config/router/default_haproxy_router.adoc (hunk 4 above)
Comment (jmencak) [inline install_config/router/default_haproxy_router.adoc]: This is correct, I've seen nodes running router pods affected with this only.  So, perhaps something like this? 

The following sysctl values are suggested for clusters with large numbers of routes on infra nodes running router pods:

Comment (jeremyeder) [inline install_config/router/default_haproxy_router.adoc]: ack

Comment (bfallonf) [inline install_config/router/default_haproxy_router.adoc]: Added a sentence on running it on the router pods. There's a chance it won't be on the infra nodes, but if so, it'll be easier to find.

File: install_config/router/default_haproxy_router.adoc [hunk 5]
```diff
@@ -1353,6 +1353,50 @@ $ oc set env dc/router -c router DROP_SYN_DURING_RESTART=true
...
+the ARP cache.
+
+The kernel messages would be similar to the following:
+
+----
+[ 1738.811139] net_ratelimit: 1045 callbacks suppressed
+[ 1743.823136] net_ratelimit: 293 callbacks suppressed
+----
+
+When using the `oc` client, the commands fail and the user receives the
```
Comment (jmencak) [inline install_config/router/default_haproxy_router.adoc]: Please remove now duplicate lines 1371-1373.

**Comment by bfallonf** [comment]
@jmencak Thanks for that. Mistake on my part.

@adellape @ahardin-rh @gaurav-nelson peer review please?

**Comment by bfallonf** [comment]
[rev_history]
|xref:../install_config/router/default_haproxy_router.adoc#install-config-router-default-haproxy[Setting up a Router -> Using the Default HAProxy Router]
|Added the xref:../install_config/router/default_haproxy_router.adoc#deploy-router--arp-cach-tuning-for-large-scale-clusters[ARP Cache Tuning for Large-scale Clusters] section.
%

**Comment by ahardin-rh** [comment]
Just 2 minor comments. Otherwise, LGTM.

**Comment by bfallonf** [comment]
Thanks @ahardin-rh . I'll merge away.

**Pull request merged**